### PR TITLE
ISPN-5922 Recovery MBean showInDoubtTransactions fails because of mis…

### PR DESCRIPTION
**Pull request by pruivo**
…sing RecoveryManager

https://issues.jboss.org/browse/ISPN-5922

### CODE REVIEW

File: core/src/main/java/org/infinispan/configuration/cache/Configurations.java [hunk 1]
```diff
@@ -66,4 +67,12 @@ public static boolean hasSharedCacheLoaderOrWriter(Configuration configuration)
       }
       return false;
    }
+
+   public static boolean isRecoveryEnabled(Configuration configuration) {
+      TransactionConfiguration txConfig = configuration.transaction();
+      return txConfig.transactionMode() == TransactionMode.TRANSACTIONAL &&  //tx cache
+            !txConfig.useSynchronization() && //xa enlistment
+            txConfig.transactionProtocol() != TransactionProtocol.TOTAL_ORDER && //non TO cache
+            txConfig.recovery().enabled(); //recovery enabled
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: We already validate the first 3 conditions in `RecoveryConfigurationBuilder`, do we need another check?

Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: well, it cross my mind that we can have a ```Configuration``` instance without validation. You could create a new instance of ```Configuration``` by hand or using ```ConfigurationBuilder.build(false)``` (skips validation)

Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: do you think that I'm overthinking?

Comment (danberindei) [inline core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: Yeah, I don't think anyone's going to use `Configurations` with a detached configuration...

Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: removed!

**Comment by wburns** [comment]
Changes seem fine to me.  @danberindei, did you have any objections?

**Comment by danberindei** [comment]
@wburns just slipped under the radar, integrating...

**Comment by danberindei** [comment]
Done, thanks Pedro!

**Pull request closed without merging**